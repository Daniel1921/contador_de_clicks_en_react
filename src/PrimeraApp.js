// import { Fragment } from "react";
import PropTypes from 'prop-types';

// funtional component

const PrimeraApp = ( {saludo, subtitulo} ) => {
    
    // const saludo = ' Hola Mundo en Espa√±ol '


    // const x = {
    //     nombre: 'Daniel',
    //     edad: 25
    // }


    return ( <>
                <h1> {saludo} !</h1>
                <p>{subtitulo}</p>
                {/* <pre>{JSON.stringify(x, null, 3)}</pre> */}
             </>
            );
}

PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}

PrimeraApp.defaultProps = {
    subtitulo: 'Componentes y propiedades en React'
}


export default PrimeraApp;