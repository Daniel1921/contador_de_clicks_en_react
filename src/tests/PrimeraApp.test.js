// import { render } from "@testing-library/react";
import PrimeraApp from "../PrimeraApp";
import {shallow} from 'enzyme'



describe('prueba en PrimeraApp', () => {
    // test('Debe de mostrar el mensaje "hola, soy daniel" ', () => {
    //     const saludo = "hola, soy daniel";
    //     const {getByText} = render(<PrimeraApp saludo="hola, soy daniel" />)
    //     expect(getByText(saludo)).toBeInTheDocument();
    // })

    test('Debe de mostrar <PrimeraApp></PrimeraApp> correctamente', () => {
        const saludo = 'hola, soy daniel'
        const wrapper = shallow(<PrimeraApp saludo={saludo} />);
        expect(wrapper).toMatchSnapshot();
    })

    test('Debe de mostrar el subtitulo enviado por props ', () => {
        const saludo = 'hola, soy daniel';
        const subtitulo = 'soy un subtitulo';
        const wrapper = shallow(<PrimeraApp 
            saludo={saludo}
            subtitulo={subtitulo} />);
        const textoParrafo = wrapper.find('p').text();
        expect(textoParrafo).toBe(subtitulo);
      
        
    })
    
    

    
})
