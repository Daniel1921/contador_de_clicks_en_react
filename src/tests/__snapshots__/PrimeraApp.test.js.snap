// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`prueba en PrimeraApp Debe de mostrar <PrimeraApp></PrimeraApp> correctamente 1`] = `
<Fragment>
  <h1>
     
    hola, soy daniel
     !
  </h1>
  <p>
    Componentes y propiedades en React
  </p>
</Fragment>
`;
